app/lib/features/pins/models/pin_edit_state/pin_edit_state.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
```

Comment:
More often than not, I've come to find the riverpod-based form-models for simple data (like here title, link, etc) to be easily broken and not working correctly (e.g. because of lacking proper updates when the focus isn't gone yet, or simply not being properly in sync at all times or ugly refreshing and/or lack of proper cursor management) and found the much simpler form+formfields-with-validators with local edit-controller to be much easier to use and less fragile.

Unless there is a secondary source for changing the provider and therefore the need to sync the data (which isn't the case here nor for many of our editors in general), what is the benefit of using this over plain old edit-controller+form-fields?

Reply:
So one benefit of using riverpod based models is that you're controlling and managing UI data with one separate class rather than having separate providers for each state variable and potentially having multiple text controllers being passed through widget tree if that variable is needed at final state stage. This would make code a bit complex and hard to refactor. And we have a lot of providers right now which are hard to track which goes where,  having riverpod state class with internal state variables mitigates a lot of noise.

Reply:
I know there are some potential problems with these models in regards of UI updates and sync problems, but to counter it we started adding UI suite which brings these issues forefront and allow us to be more helpful in investigating.